Create strings.xml files in the Gradle project to handle localization

amanj120 opened the pull request:
This PR is a forward port of #40182

Calinou committed:
Mention C# gotchas in Object's dynamic call/set/connect methods

This closes #34015.

Faless committed:
UDPServer handles PacketPeerUDP-client association

UDPServer now uses a single socket which is shared with the
PacketPeerUDP it creates and has a new `poll` function to read incoming
packets on that socket and delivers them to the appropriate peer.
PacketPeerUDP created this way never reads from the socket, but are
allowed to write on it using sendto.

This is needed because Windows (unlike Linux/BSD) does not support
packet routing when multiple sockets are bound on the same address/port.

Commit:
Create strings.xml files in the gradle project to handle localization

Commit:
Create strings.xml files in the gradle project to handle localization

danielzting committed:
Fix cancel/OK button order on macOS

The macOS platform convention regarding button order is cancel on left,
OK on right.

stijn-h committed:
Resize dialogs (FileDialog, EditorFileDialog, Reparent, SceneTreeDialog and resource depency dialogs).

Calinou committed:
Optimize the default iOS splash images

Command used:

    parallel oxipng -o6 --strip --zopfli ::: **/*.png

This saves about 14 KB in total.

Calinou committed:
Disable file logging for the project manager

Due to `user://` returning the current working directory when no
project is open, this caused logs to be written to `$HOME`
most of the time.

This closes #40305.

Faless committed:
Document updated UDPServer interface.

Faless committed:
Funnel refuse_new_connections to Godot ENet.

Faless committed:
Reorganize ENet pactches.

stijn-h committed:
Improve quick open performance and update some behaviour

 - Cache possible files and icons at popup
 - Sort files with heapsort instead of selection sort

 - Always scroll back to top (selection) upon refresh
 - Scoring function: fix second case and remove expensive similarity computation for insignificant results.
 - Only show a max amount of files (currently set at 300)

akien-mga commented on platform/android/export/export.cpp:
```diff
@@ -29,6 +29,7 @@
 /*************************************************************************/
 
 #include "export.h"
+#include "gradle_export_util.h"
```
This should be in the block below. Only the file's corresponding header is listed separately (`export.h` for `export.cpp`): https://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html#header-includes

akien-mga commented on platform/android/java/app/res/values-fa/godot_project_name_string.xml:
```diff
@@ -0,0 +1,6 @@
+<resources>
+    <string name="godot_project_name_string">godot-project-name-fa</string>
+</resources>
+
```
There's an extra empty line here.

akien-mga commented on platform/android/java/app/res/values-in/godot_project_name_string.xml:
```diff
@@ -0,0 +1,6 @@
+<resources>
+    <string name="godot_project_name_string">godot-project-name-in</string>
+</resources>
+
```
Here too.

akien-mga commented on platform/android/java/app/res/values-iw/godot_project_name_string.xml:
```diff
@@ -0,0 +1,6 @@
+<resources>
+    <string name="godot_project_name_string">godot-project-name-iw</string>
+</resources>
+
```
And here.

akien-mga commented on platform/android/java/app/res/values-ko/godot_project_name_string.xml:
```diff
@@ -0,0 +1,6 @@
+<resources>
+    <string name="godot_project_name_string">godot-project-name-ko</string>
+</resources>
+
```
And here.

akien-mga commented on platform/android/java/app/res/values-zh-rHK/godot_project_name_string.xml:
```diff
@@ -1,4 +1,6 @@
 <resources>
     <string name="godot_project_name_string">godot-project-name-zh_HK</string>
 </resources>
+
```
And here.

akien-mga commented on platform/android/java/app/res/values-zh-rTW/godot_project_name_string.xml:
```diff
@@ -1,4 +1,6 @@
 <resources>
     <string name="godot_project_name_string">godot-project-name-zh_TW</string>
 </resources>
+
```
And here.

akien-mga commented on platform/android/java/app/res/values-zh/godot_project_name_string.xml:
```diff
@@ -1,4 +1,6 @@
 <resources>
     <string name="godot_project_name_string">godot-project-name-zh</string>
 </resources>
+
```
And here.

akien-mga commented on platform/android/java/app/res/values/godot_project_name_string.xml:
```diff
@@ -0,0 +1,6 @@
+<resources>
+    <string name="godot_project_name_string">godot-project-name</string>
+</resources>
+
```
And here.

akien-mga committed:
SCons: Do not enable werror=yes by default

There are too many users who compile Godot from source and are not familiar
with the buildsystem or C/C++ compilation warnings, and thus report any kind
of yet-unfixed warning as a (often duplicate) bug.

Compiler warnings change at every compiler version and are different for each
compiler, so it's difficult to ensure that the codebase would always be 100%
warning-free, especially in the future.

I already disabled it for stable releases in #37958, but having it on non
stable commits could also become an annoyance in the future when trying to
bisect issues with a new compiler version which emits warnings unknown at
the time of commit.

TL;DR: Contributors, use `dev=yes` or `werror=yes`. CI does and won't let you
create new warnings ;)

rcorre committed:
Revert "Include gdscript warning name in the warning message."

This reverts commit de3ad3b30ecb8de1aa112df7d61630102f077b5b.

rcorre committed:
Include gdscript warning name in LSP message.

My initial attempt changed this in the gdscript code, which resulted in
a duplicate warning name in the builtin editor. We should just append
the warning name in the LSP instead.

This uses parens to match what is shown in the builtin editor.

akien-mga committed:
PO loader: Fix unclosed files and error messages

Trying to get `f->get_path()` after deleting `f` was not super clever :)

Fixes #40324.

bruvzg committed:
[macOS] Fix window size on macOS Big Sur (title bar height is no longer same as menu height), use top-left corner as resize origin instead of bottom-left.

Paulb23 committed:
Fixed visual shader editor not clearing colour regions

reduz committed:
Properly pass safe margin on initialization.

Fixes jitter.

Calinou committed:
Document VehicleBody3D and VehicleWheel3D limitations

These classes have dozens of open bugs and missing features
which may not be fixed anytime soon.

It's probably better to document it upfront at this point.

akien-mga committed:
Merge pull request #40367 from akien-mga/scons-disable-werror

SCons: Do not enable werror=yes by default

akien-mga committed:
Merge pull request #40371 from rcorre/warning-fix

Show gdscript warning in LSP without duplication.

akien-mga committed:
Merge pull request #40372 from akien-mga/close-those-po-or-files

PO loader: Fix unclosed files and error messages

akien-mga committed:
Merge pull request #40374 from Faless/udp/server_abstraction

UDPServer uses single socket, abstract clients.

akien-mga committed:
Merge pull request #40375 from Paulb23/fix_visual_shader_clearing_color_regions

Fixed visual shader editor not clearing colour regions

akien-mga committed:
Merge pull request #40315 from Calinou/project-manager-disable-file-logging

Disable file logging for the project manager

touilleMan committed:
Correct RichTextLabel.custom_effects property type metadata

akien-mga committed:
Merge pull request #40379 from bruvzg/macos_11_window_size

[macOS] Fix window size on macOS Big Sur, use top-left corner as resize origin.

touilleMan committed:
Correct is_reference attribute in api.json for Reference class

thebestnom committed:
Android: fix display server always alerting no Vulkan support

Chaosus committed:
Fix some incorrect conversions which lead to crash in shaders

akien-mga committed:
Merge pull request #40377 from reduz/fix-default-kb3d-safe-margin

Properly pass safe margin on initialization.

akien-mga committed:
Merge pull request #40298 from Calinou/optimize-ios-splash

Optimize the default iOS splash images

touilleMan committed:
Add missing has_default_value field for signals in api.json

Chaosus committed:
Merge pull request #40389 from Chaosus/fix_shader

 Fix some incorrect conversions which lead to crash in shaders

akien-mga committed:
Merge pull request #40392 from touilleMan/missing-has_default_value-signals-api_json

Add missing has_default_value field for signals in api.json

akien-mga committed:
Merge pull request #40387 from thebestnom/Android-Fix-Display-Server

Android: fix display server always alerting no Vulkan support

akien-mga committed:
Merge pull request #40383 from touilleMan/correct-RichTextLabel.custom_effects-property-type

Correct RichTextLabel.custom_effects property type metadata

akien-mga committed:
Merge pull request #40386 from touilleMan/correct-Reference-is_reference-api.json

Correct is_reference attribute in api.json for Reference class

akien-mga committed:
doc: Sync classref with current source

akien-mga committed:
Merge pull request #40268 from DanielZTing/master

Fix cancel/OK button order on macOS

akien-mga committed:
Merge pull request #40380 from Calinou/doc-vehiclebody-limitations

Document VehicleBody3D and VehicleWheel3D limitations

Calinou committed:
Document which escape sequences are supported by `String.c_unescape()`

See https://github.com/godotengine/godot/issues/38716.

akien-mga committed:
Merge pull request #40376 from hinlopen/quick-open-improvements

Improve quick open performance and behaviour

akien-mga committed:
Merge pull request #40291 from hinlopen/dialog-size

Resize various dialogs

akien-mga committed:
Merge pull request #37961 from Calinou/doc-csharp-dynamic-object-call

Mention C# gotchas in Object's dynamic call/set/connect methods

akien-mga commented:
There seems to be conflicts now with the last update. Otherwise it should be good to merge.

rcorre committed:
Clarify how to convert PrimitiveMesh to ArrayMesh.

It took me a bit to figure this out, as I was initially doing something
more complicated like this before I realized I just had to pass
get_mesh_arrays directly to add_surface_from_arrays.

```
var arr_mesh = ArrayMesh.new()
var arrays = []
arrays.resize(ArrayMesh.ARRAY_MAX)
arrays[ArrayMesh.ARRAY_VERTEX] = c.get_mesh_arrays()
arr_mesh.add_surface_from_arrays(Mesh.PRIMITIVE_TRIANGLES, arrays)
```

akien-mga committed:
Merge pull request #40412 from Calinou/doc-string-c-unescape

Document which escape sequences are supported by `String.c_unescape()`

akien-mga committed:
Merge pull request #40414 from rcorre/get-mesh-arrays-doc

Clarify how to convert PrimitiveMesh to ArrayMesh.

akien-mga commented on platform/android/export/export.cpp:
```diff
@@ -43,6 +43,7 @@
 #include "editor/editor_log.h"
 #include "editor/editor_node.h"
 #include "editor/editor_settings.h"
+#include "platform/android/export/gradle_export_util.h"
```
Is it expected that this header is used in this PR, but not in #40182?

amanj120 replied:
It will be used in #40182. The difference is that (up until about 15 minutes ago), the 3.2 branch did not have the `gradle_export_util.h` file

amanj120 committed:
added newlines to xml files to fix merge conflict

amanj120 committed:
added newlines to xml files to fix merge conflict

amanj120 committed:
Merge branch 'forward_port_bundle_pr_resources' of https://github.com/amanj120/godot into forward_port_bundle_pr_resources

amanj120 committed:
final touchup for the merge conflict, hopefully this triggers an update to the PR

amanj120 commented:
I am not sure why there is a merge conflict on this PR. I fixed the changes on [my branch](https://github.com/godotengine/godot/compare/master...amanj120:forward_port_bundle_pr_resources) but the PR isn't reflecting those changes.

amanj120 commented:
Something went wrong when I tried to fix the merge conflict
